### PR TITLE
Multi output crafter

### PR DESCRIPTION
**Pull request by MEEPofFaith**
Adds `outputItems` to `GenericCrafter`. Input an array of ItemStacks and upon craft will produce everything in the array.

`outputItem` still exists so as not to break stuff.

### CODE REVIEW

**Comment by BlueWolf3682** [comment]
comically large sized thank

**Comment by MrDuck557** [comment]
deprecate `outputItem` to make it clear that you shouldn't use it
This opens up possibilities

**Comment by MEEPofFaith** [comment]
> deprecate `outputItem` to make it clear that you shouldn't use it
> This opens up possibilities

~~I don't know how lol~~

**Comment by notMEEPoffaith** [comment]
heh so uh i started this with a few pains in making a box unpacker have multiple outputs

File: core/src/mindustry/world/blocks/production/GenericCrafter.java [hunk 1]
```diff
@@ -15,7 +15,9 @@
 import mindustry.world.meta.*;
 
 public class GenericCrafter extends Block{
+    /** @deprecated use outputItems instead, outputItem would likely be removed eventually **/
     public @Nullable ItemStack outputItem;
```
Comment (Anuken) [inline core/src/mindustry/world/blocks/production/GenericCrafter.java:19]: `outputItem` is too convenient and too widely used to deprecated. Instead, use **only** `outputItems` in the update code, and make `outputItems` a single-item array if it is null and `outputItems != null` in `init()`.

**Comment by notMEEPoffaith** [comment]
un-deprecate output item

**Comment by notMEEPoffaith** [comment]
and the cat will merge basically

File: core/src/mindustry/world/blocks/production/GenericCrafter.java (hunk 1 above)
Comment (Anuken) [reply core/src/mindustry/world/blocks/production/GenericCrafter.java:19]: Also worth noting that this implementation doesn't handle `shouldConsume` properly.

Comment (murtaza64) [reply core/src/mindustry/world/blocks/production/GenericCrafter.java:19]: Isn't `outputsItems` also not handled correctly?

Comment (Anuken) [reply core/src/mindustry/world/blocks/production/GenericCrafter.java:19]: yes

**Comment by MEEPofFaith** [comment]
Done

File: core/src/mindustry/world/blocks/production/GenericCrafter.java [hunk 2]
```diff
@@ -133,15 +141,26 @@ public void updateTile(){
...
+                    }
+                }
             }
 
             if(outputLiquid != null){
                 dumpLiquid(outputLiquid.liquid);
             }
         }
 
+        public void produceItem(ItemStack item){
```
Comment (Anuken) [inline core/src/mindustry/world/blocks/production/GenericCrafter.java:158]: This should be inlined, since it's only used in one location now.

**Comment by Anuken** [comment]
Merged manually.

**Comment by MEEPofFaith** [comment]
k

**Pull request closed without merging**